File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        
        char[] vowels = {'a','e','i','o','u','A','E','I','O','U'};
        char[] sArray = s.toCharArray();
        
        
        int left = 0;
        int right = s.length()-1;
        
        while(left < right) {
            
            while(left<s.length()-1 && !isVowel(sArray[left], vowels)) {
                left++;
            }
            
            while(right>=0 && !isVowel(sArray[right], vowels)) {
                right--;
            }
            
            if(left< right){
                
                char temp = sArray[left];
                sArray[left] = sArray[right];
                sArray[right]= temp;
                left++;
                right--;
            }  
        }
        
        return new String(sArray);
             
    }
    
    private boolean isVowel(char c, char[] vowels) {
        for(char v: vowels){
            if(v==c) {
                return true;
            }
        }
        return false;
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        
        left =0
        right = len(s)-1
        
        vowels = ['a','e','i','o','u','A','E','I','O','U']
        s = list(s)
        
        while left<right:
            
            while left<len(s)-1 and s[left] not in vowels:
                left+=1
            
            while right>=0 and s[right] not in vowels:
                right-=1
                
            if left < right:
                s[left], s[right] = s[right], s[left]
                left+=1
                right-=1
        
        
        return "".join(s)
        